File: create_tables/field.py
import oracledb
import sys
import lib_platform as platform

if platform.system == "darwin":
    sys.path.insert(0, "database")
elif platform.system == "windows":
    sys.path.insert(0, "../database")

from conn import connection

# pw = getpass.getpass("Enter password: ")

# connection = oracledb.connect(
#     user="admin",
#     password="ppdm-3.9",
#     dsn="/kei-oracle-test.chkxr21l2l5s.ap-southeast-3.rds.amazonaws.com:1521/ORCL")

# print("Successfully connected to Oracle Database")

cursor = connection.cursor()


cursor.execute("""
    begin
        execute immediate 'drop table field';
        exception when others then if sqlcode <> -942 then raise; end if;
    end;""")

cursor.execute("""
    create table field(
    FIELD_ID VARCHAR(40) PRIMARY KEY NOT NULL,
    ACTIVE_IND VARCHAR(1),
    AREA_ID VARCHAR(40),
    AREA_TYPE VARCHAR(40),
    DISCOVERY_DATE TIMESTAMP(0),
    EFFECTIVE_DATE TIMESTAMP(0),
    EXPIRY_DATE TIMESTAMP(0),
    FIELD_ABBREVIATION VARCHAR(12),
    FIELD_AREA_OBS_NO INT,
    FIELD_NAME VARCHAR(255),
    FIELD_TYPE VARCHAR(40),
    GROUP_FIELD_ID VARCHAR(40),
    PPDM_GUID VARCHAR(38),
    REMARK VARCHAR(2000),
    SOURCE VARCHAR(40),
    ROW_CHANGED_BY VARCHAR(30),
    ROW_CHANGED_DATE TIMESTAMP(0),
    ROW_CREATED_BY VARCHAR(30),
    ROW_CREATED_DATE TIMESTAMP(0),
    ROW_EFFECTIVE_DATE TIMESTAMP(0),
    ROW_EXPIRY_DATE TIMESTAMP(0),
    ROW_QUALITY VARCHAR(40)
    )""")

File: postgres_39/struct/Field.go
package dto

import (
	"time"
)

type Field struct {
	Field_id           string     `json:"field_id"`
	Active_ind         *string    `json:"active_ind"`
	Area_id            *string    `json:"area_id"`
	Area_type          *string    `json:"area_type"`
	Discovery_date     *time.Time `json:"discovery_date"`
	Effective_date     *time.Time `json:"effective_date"`
	Expiry_date        *time.Time `json:"expiry_date"`
	Field_abbreviation *string    `json:"field_abbreviation"`
	Field_area_obs_no  *int       `json:"field_area_obs_no"`
	Field_name         *string    `json:"field_name"`
	Field_type         *string    `json:"field_type"`
	Group_field_id     *string    `json:"group_field_id"`
	Ppdm_guid          string     `json:"ppdm_guid"`
	Remark             *string    `json:"remark"`
	Source             *string    `json:"source"`
	Row_changed_by     *string    `json:"row_changed_by"`
	Row_changed_date   *time.Time `json:"row_changed_date"`
	Row_created_by     *string    `json:"row_created_by"`
	Row_created_date   *time.Time `json:"row_created_date"`
	Row_effective_date *time.Time `json:"row_effective_date"`
	Row_expiry_date    *time.Time `json:"row_expiry_date"`
	Row_quality        *string    `json:"row_quality"`
}
